### PR TITLE
[Workers AI] Restructure models and add new models

### PR DESCRIPTION
**Pull request by craigsdennis**
This replaces the previous /workers-ai/models page which had Task Type Categories to be each model having it's own URL/permalink.

This adds a new layout and a new way of including optional partial overrides.

The content pages are built from the `https://api.cloudflare.com/client/v4/accounts/${process.env.AUTH_ACCOUNT}/ai/models/search` API. They will be automatically generated going forward.

Please do not merge before planned new model launch on 2/28/24.

### CODE REVIEW

**Comment by markdembo** [comment]
This PR introduces the following changes:

- Every model gets its own page instead of being grouped by task type
- Added 20+ new models
- Added a new layout for each model detail page, that is easily extensible with ability to override parts if needed
- Introduces new patterns for easy addition of new models using scripts (external; not part of this PR)

Formalities taken care off:
- Added redirects
- Fixed broken links

Pending issues, to be completed by @craigsdennis after this commit (see future commits):

1️⃣ Fix tiny formatting error ( `## Code Examples` does not get converted to h2 headline on model detail page)
2️⃣ Add impainting example

While fixing broken links, Prettier ran and changes some formatting. I did not see any breaking changes in the related doc sections.

Please @kodster28 take a look if you have time. I will work with @deadlypants1973 tomorrow to get any outstanding issues fixed.

**Review by kodster28 — APPROVED** [review]
Weird... the way to solve the `## Code Examples` rendering seems to be just to have a new line at the start of the file.

@craigsdennis pointed this out to me earlier + still can't make heads or tails of it... but that's sometimes Hugo for you.

Approved from a `This doesn't break the docs site + renders fine perspective.` Also, generally looks great and super impressed by the architecture. Thanks for y'alls hard work on this.

File: layouts/partials/models/code-examples/automatic-speech-recognition.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  },
+};
+```
+
+</details>
+
+<details>
+  <summary>curl</summary>
+
+```sh
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/automatic-speech-recognition.md]:
````suggestion
```bash
````
We use `bash` for multi-line `curl` commands. If we keep `sh`, we **must** add a `$` prefix to the first line of the command.

File: layouts/partials/models/code-examples/image-classification.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  },
+};
+```
+
+</details>
+
+<details>
+  <summary>curl</summary>
+
+```sh
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/image-classification.md]:
```suggestion
```bash
```

File: layouts/partials/models/code-examples/object-detection.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  },
+};
+```
+
+</details>
+
+<details>
+  <summary>curl</summary>
+
+```sh
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/object-detection.md:39]:
```suggestion
```bash
```

File: layouts/partials/models/code-examples/stable-diffusion-v1-5-img2img.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+};
+```
+
+</details>
+
+<details>
+  <summary>curl</summary>
+
+```sh
+$ curl https://api.cloudflare.com/client/v4/accounts/$CLOUDFLARE_ACCOUNT_ID/ai/run/{{ .Page.Params.model.name }} \
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/stable-diffusion-v1-5-img2img.md]:
````suggestion
```bash
curl https://api.cloudflare.com/client/v4/accounts/{account_id}/ai/run/{{ .Page.Params.model.name }} \
````
We use `bash` for multi-line `curl` commands. Removing the `$` prefix to make it consistent with other `curl` commands.

File: layouts/partials/models/code-examples/stable-diffusion-v1-5-inpainting.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+};
+```
+
+</details>
+
+<details>
+  <summary>curl</summary>
+
+```sh
+$ curl https://api.cloudflare.com/client/v4/accounts/$CLOUDFLARE_ACCOUNT_ID/ai/run/{{ .Page.Params.model.name }} \
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/stable-diffusion-v1-5-inpainting.md]:
````suggestion
```bash
curl https://api.cloudflare.com/client/v4/accounts/{account_id}/ai/run/{{ .Page.Params.model.name }} \
````

File: layouts/partials/models/code-examples/text-classification.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+print(output)
+```
+
+</details>
+
+<details>
+  <summary>curl</summary>
+
+```sh
+curl https://api.cloudflare.com/client/v4/accounts/$CLOUDFLARE_ACCOUNT_ID}/ai/run/{{ .Page.Params.model.name }} \
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/text-classification.md]:
````suggestion
```bash
curl https://api.cloudflare.com/client/v4/accounts/{account_id}/ai/run/{{ .Page.Params.model.name }} \
````

File: layouts/partials/models/code-examples/text-embeddings.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+print(response.json())
+```
+
+</details>
+
+<details>
+  <summary>curl</summary>
+
+```sh
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/text-embeddings.md]:
````suggestion
```bash
````

File: layouts/partials/models/code-examples/text-generation.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+result = response.json()
+return result.response
+```
+
+</details>
+
+<details>
+  <summary>curl</summary>
+
+```sh
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/text-generation.md]:
````suggestion
```bash
````

File: layouts/partials/models/code-examples/text-to-image.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+};
+```
+
+</details>
+
+<details>
+  <summary>curl</summary>
+
+```sh
+$ curl https://api.cloudflare.com/client/v4/accounts/$CLOUDFLARE_ACCOUNT_ID/ai/run/{{ .Page.Params.model.name }} \
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/text-to-image.md]:
````suggestion
```bash
curl https://api.cloudflare.com/client/v4/accounts/{account_id}/ai/run/{{ .Page.Params.model.name }} \
````

File: layouts/partials/models/code-examples/translation.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+print(output)
+```
+
+</details>
+
+<details>
+  <summary>curl</summary>
+
+```sh
+curl https://api.cloudflare.com/client/v4/accounts/$CLOUDFLARE_ACCOUNT_ID}/ai/run/{{ .Page.Params.model.name }} \
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/translation.md]:
````suggestion
```bash
curl https://api.cloudflare.com/client/v4/accounts/{account_id}/ai/run/{{ .Page.Params.model.name }} \
````

File: layouts/partials/models/responses/text-generation.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+  headers: {
+    "content-type": "text/event-stream",
+  },
+});
+```
+
+Using the REST API:
+
+```sh
+$ curl https://api.cloudflare.com/client/v4/accounts/{ACCOUNT_ID}/ai/run/@cf/meta/llama-2-7b-chat-int8 \
```
Comment (pedrosousa) [inline layouts/partials/models/responses/text-generation.md:32]:
```suggestion
```bash
curl https://api.cloudflare.com/client/v4/accounts/{account_id}/ai/run/@cf/meta/llama-2-7b-chat-int8 \
```
Unless we're using a previously defined variable, our placeholder format for URLs is `{account_id}`.

File: layouts/partials/models/responses/text-generation.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+  -X POST \
+  -H "Authorization: Bearer {API_TOKEN}" \
+  -d '{ "stream": true, "messages": [{ "role": "system", "content": "You are a friendly assistant" }, { "role": "user", "content": "Why is pizza so good?" }]}'
+```
+
+Streaming responses use [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events); the are easy to use, simple to implement on the server side, standardized, and broadly available across many platforms natively or as a polyfill.
+
+```sh
+$ curl -X POST \
+"https://api.cloudflare.com/client/v4/accounts/<account>/ai/run/@cf/meta/llama-2-7b-chat-int8" \
```
Comment (pedrosousa) [inline layouts/partials/models/responses/text-generation.md:42]:
```suggestion
```bash
curl -X POST \
"https://api.cloudflare.com/client/v4/accounts/{account_id}/ai/run/@cf/meta/llama-2-7b-chat-int8" \
```

File: layouts/partials/models/code-examples/automatic-speech-recognition.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+curl https://api.cloudflare.com/client/v4/accounts/$CLOUDFLARE_ACCOUNT_ID}/ai/run/{{ .Page.Params.model.name }} \
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/automatic-speech-recognition.md]:
```suggestion

```bash
curl https://api.cloudflare.com/client/v4/accounts/{account_id}/ai/run/{{ .Page.Params.model.name }} \
```
Our [Style Guide](https://developers.cloudflare.com/style-guide/api-content-strategy/guidelines-for-curl-commands/) contains some guidance on writing `curl` commands in documentation.

File: layouts/partials/models/code-examples/automatic-speech-recognition.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+  -X POST \
+  -H "Authorization: Bearer $CLOUDFLARE_API_TOKEN" \
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/automatic-speech-recognition.md:44]:
```suggestion
  -H "Authorization: Bearer <API_TOKEN>" \
```
This change would also apply to other occurrences.

File: layouts/partials/models/code-examples/image-classification.md [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+curl https://api.cloudflare.com/client/v4/accounts/$CLOUDFLARE_ACCOUNT_ID/ai/run/{{ .Page.Params.model.name }} \
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/image-classification.md:40]:
```suggestion
curl https://api.cloudflare.com/client/v4/accounts/$CLOUDFLARE_ACCOUNT_ID/ai/run/{{ .Page.Params.model.name }} \
```
```suggestion
curl https://api.cloudflare.com/client/v4/accounts/{account_id}/ai/run/{{ .Page.Params.model.name }} \
```

File: layouts/partials/models/code-examples/image-classification.md [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+    -X POST \
+    -H "Authorization: Bearer $CLOUDFLARE_API_TOKEN" \
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/image-classification.md:42]:
```suggestion
    -H "Authorization: Bearer <API_TOKEN>" \
```

File: layouts/partials/models/code-examples/object-detection.md [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+curl https://api.cloudflare.com/client/v4/accounts/$CLOUDFLARE_ACCOUNT_ID/ai/run/@cf/meta/detr-resnet-50 \
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/object-detection.md:40]:
```suggestion
curl https://api.cloudflare.com/client/v4/accounts/{account_id}/ai/run/@cf/meta/detr-resnet-50 \
```

File: layouts/partials/models/code-examples/text-embeddings.md [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+curl https://api.cloudflare.com/client/v4/accounts/$CLOUDFLARE_ACCOUNT_ID/ai/run/{{ .Page.Params.model.name }} \
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/text-embeddings.md:71]:
```suggestion
curl https://api.cloudflare.com/client/v4/accounts/{account_id}/ai/run/{{ .Page.Params.model.name }} \
```

File: layouts/partials/models/code-examples/text-generation.md [hunk 19]
```diff
@@ -0,0 +1,110 @@
...
+curl https://api.cloudflare.com/client/v4/accounts/$CLOUDFLARE_ACCOUNT_ID/ai/run/{{ .Params.model.name }} \
```
Comment (pedrosousa) [inline layouts/partials/models/code-examples/text-generation.md:104]:
```suggestion
curl https://api.cloudflare.com/client/v4/accounts/{account_id}/ai/run/{{ .Params.model.name }} \
```

File: layouts/partials/models/code-examples/translation.md (hunk 10 above)
Comment (craigsdennis) [reply layouts/partials/models/code-examples/translation.md]: Why switch out the env variable? Was trying to make it copy pasteable

Comment (pedrosousa) [reply layouts/partials/models/code-examples/translation.md]: Unless we provide the variable definition somewhere in the same block (or the same page), we tend to use a placeholder and let users switch to env variables if they want to do so.

If we want to keep the variable nevertheless, let's fix the typo (extra `}`).

Comment (craigsdennis) [reply layouts/partials/models/code-examples/translation.md]: I think we should, I can't find this in the style guide. We don't include instructions on replacing the {account_id} and it looks like a template. Probably not the best PR for this discussion. Will bring this up in a chat. and apply your changes.

Thanks for the catch!

File: layouts/partials/models/code-examples/automatic-speech-recognition.md (hunk 1 above)
Comment (craigsdennis) [reply layouts/partials/models/code-examples/automatic-speech-recognition.md]: Fixed

**Pull request merged**